sql/moz-fx-data-shared-prod/firefox_ios_derived/app_store_retention_week_4_v1/query.sql

```diff
@@ -0,0 +1,52 @@
+WITH clients_retention AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    days_seen_bits,
+  FROM
+    firefox_ios.baseline_clients_last_seen
+  WHERE
+    submission_date = @submission_date
+    AND normalized_channel = "release"
+),
+first_seen AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+  FROM
+    firefox_ios.baseline_clients_first_seen
+  -- 28 days need to elapse before calculating the week 4 and day 28 retention metrics
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 27 DAY)
+),
+retention_calculation AS (
+  SELECT
+    first_seen.first_seen_date,
+    clients_retention.client_id,
+    clients_retention.sample_id,
+    LENGTH(REPLACE(mozfun.bits28.to_string(days_seen_bits), "0", "")) AS days_seen_in_first_28_days,
+    mozfun.bits28.retention(days_seen_bits, @submission_date) AS retention,
+  FROM
+    clients_retention
+  INNER JOIN
+    first_seen
+  USING
+    (client_id, sample_id)
+)
+SELECT
+  * EXCEPT (retention) REPLACE(
+    -- metric date should align with first_seen_date, if that is not the case then the query will fail.
+    IF(
+      retention.day_27.metric_date <> first_seen_date,
+      ERROR("Metric date misaligned with first_seen_date"),
+      first_seen_date
+    ) AS first_seen_date
+  ),
+  days_seen_in_first_28_days > 1 AS repeat_first_month_user,
+  -- added for testing.
+  retention.day_27.active_in_week_1 AS retained_week_2,
+  retention.day_27.active_in_week_3 AS retained_week_4,
```

Comment:
This is correct. This is saying "For users who started 27 days ago, which were active on days 21-27 (0-indexed)?"